### PR TITLE
[config] Support to modify configs when BE is running without restarting

### PR DESCRIPTION
**Pull request by acelyc111**
Ref https://github.com/apache/incubator-doris/issues/3268
In the past, when we want to modify some BE configs, we have to modify be.conf and then restart BE.
This patch provides a way to modify configs in the type of 'threshold', 'interval', 'enable flag'
when BE is running without restarting it.
You can update a single config once by BE's http API: `be_host:be_http_port/api/update_config?config_name=new_value`

### CODE REVIEW

**Comment by imay** [comment]
@acelyc111 This is a wonderful job, could you please to create an issue to track this feature and relate this PR to it?

File: be/src/http/action/update_config_action.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    LOG(INFO) << req->debug_string();
+
+    bool success = true;
+    std::string msg;
+    if (req->params()->size() != 1) {
+        success = false;
+        msg = "Now only support to set a single config once, via 'config_name=new_value'.";
+    } else {
+        DCHECK(req->params()->size() == 1);
+        for (const auto& it : *req->params()) {
```
Comment (imay) [inline be/src/http/action/update_config_action.cpp:51]: If number of params is 1, using the for statement can be misleading.

File: be/src/http/action/update_config_action.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            Status s = config::set_config(it.first, it.second);
+            success = s.ok();
+            LOG(WARNING) << "set_config " << it.first << "=" << it.second
```
Comment (imay) [inline be/src/http/action/update_config_action.cpp:54]: when `s.ok()` is true, print INFO log; when `s.ok()` is false, print waring log

File: be/src/common/configbase.cpp [hunk 3]
```diff
@@ -28,91 +28,19 @@
...
 
+#include "common/status.h"
+#include "gutil/strings/substitute.h"
+
 namespace doris {
 namespace config {
 
-std::list<Register::Field>* Register::_s_fieldlist = nullptr;
-std::map<std::string, std::string>* confmap = nullptr;
+std::map<std::string, Register::Field>* Register::_s_fieldlist = nullptr;
```
Comment (imay) [inline be/src/common/configbase.cpp:37]:
```suggestion
std::map<std::string, Register::Field>* Register::_s_field_map = nullptr;
```

**Comment by acelyc111** [comment]
> @acelyc111 This is a wonderful job, could you please to create an issue to track this feature and relate this PR to it?

OK, https://github.com/apache/incubator-doris/issues/3268

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**